planet/auth.py

```diff
@@ -0,0 +1,285 @@
+        '''
+        variable_name = variable_name or ENV_API_KEY
+        api_key = os.getenv(variable_name)
+        try:
+            auth = APIKeyAuth(api_key)
+            LOGGER.debug(f'Auth set from environment variable {variable_name}')
+        except APIKeyAuthException:
+            raise AuthException(
+                f'Environment variable {variable_name} either does not exist '
+                'or is empty.')
+        return auth
+
+    @staticmethod
+    def from_login(
+        email: str,
+        password: str,
+        base_url: str = None
+    ) -> Auth:
+        '''Create authentication from login email and password.
+
+        Note: To keep your password secure, the use of `getpass` is
+        recommended.
+
+        Parameters:
+            email: Planet account email address.
+            password: Planet account password.
+            base_url: The base URL to use. Defaults to production
+                authentication API base url.
+        '''
+        cl = AuthClient(base_url=base_url)
+        auth_data = cl.login(email, password)
+
+        api_key = auth_data['api_key']
+        auth = APIKeyAuth(api_key)
+        LOGGER.debug('Auth set from login email and password')
+        return auth
+
+    @classmethod
+    @abc.abstractmethod
+    def from_dict(
+        cls,
+        data: dict
+    ) -> Auth:
+        pass
+
+    @property
+    @abc.abstractmethod
+    def value(self):
+        pass
+
+    def write(
+        self,
+        filename: str = None
+    ):
+        '''Write authentication information.
+
+        Parameters:
+            filename: Alternate path for the planet secret file.
+        '''
+        filename = filename or SECRET_FILE_PATH
+        secret_file = _SecretFile(filename)
+        secret_file.write(self.to_dict())
+
+
+class AuthClient():
+    def __init__(
+        self,
+        base_url: str = None
+    ):
+        """
+        Parameters:
+            base_url: Alternate authentication api base URL.
+        """
+        self._base_url = base_url or AUTH_URL
+        if not self._base_url.endswith('/'):
+            self._base_url += '/'
+
+    def login(
+        self,
+        email: str,
+        password: str
+    ) -> dict:
+        '''Login using email identity and credentials.
+
+        Note: To keep your password secure, the use of `getpass` is
+        recommended.
+
+        Parameters:
+            email: Planet account email address.
+            password:  Planet account password.
+
+        Returns:
+             A JSON object containing an `api_key` property with the user's
+        API_KEY.
+        '''
+        url = self._base_url + 'login'
+        data = {'email': email,
+                'password': password
+                }
+
+        sess = http.AuthSession()
+        req = models.Request(url, method='POST', json=data)
+        resp = sess.request(req)
+        auth_data = self.decode_response(resp)
+        return auth_data
```

Comment:
Eventually you may want to consider keeping the full JWT around since various services are moving toward using JWT auth headers over API key headers. Realistically it will likely be a while before everything supports JWT auth headers, though.

Reply:
right now we are returning what `jwt.decode()` returns, which is the body of the JWT. That body contains more than the api key tho that's all we are using atm.